### DISCUSSION TITLE
Basic guide to setting up kibot in windows with WSL

### DISCUSSION THREAD

**Discussion by dbriscoe511** [INTI-CMNB/KiBot#264] [Other platforms]
**1. Install WSL**
There are many guides on how to do this, but if you are on a modern version of windows, simply run in CMD as admin:

`wsl.exe --install`
`dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart`
`dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart`

Restart your PC
Next, pick your distibution of choice. I used [Ubuntu LTS 20.04](https://www.microsoft.com/store/productId/9MTTCL66CPXJ) for simplicity 
In CMD as admin run:
`
wsl --set-default-version 2
`
Restart your PC
**2. Install software to run Kibot** 
Open the WSL terminal and run:

`sudo apt update`
`sudo apt upgrade`
install kicad
`sudo add-apt-repository --yes ppa:kicad/kicad-6.0-releases`
`sudo apt update`
`sudo apt install --install-recommends kicad`
install kibot and kiauto (updated with setsoft suggestions)
`$ wget https://set-soft.github.io/debian/kibot.list`
`$ sudo cp kibot.list /etc/apt/sources.list.d/`
`$ wget https://set-soft.github.io/debian/kibot.gpg`
`$ sudo cp kibot.gpg /etc/apt/`
`$ sudo apt-get update`
`$ sudo apt-get install kibot`

**3. Copy anything you need over from your windows environment to the Linux environment**

- copy Kicad environment variables (if you have anything custom in kicad->Preferences->configure paths), such as library references that are not project specific.
               - Windows stores these variables in C:\Users\username\AppData\Roaming” + kicad\ (= %appdata%\kicad), WSL stores them in ~/.config/kicad/  just copy the environment section of the JSON between these files

**4. Running kibot**

Once WSL is installed, the Shift+Rightclick menu will show "open linux terminal here" as an option. Once you are in the linux terminal in the correct directory kibot can be run as normal. Alternativly, in powersheel/cmd/git bash for windows run 
`wsl kibot`

**5.Known issues/troubleshooting**

- Netlist output intermittently does not work, returns:
`
ERROR:Unknown KiCad dialog: KiCad PCB Editor Information (pcbnew_do - interposer.py:362) (kibot - kiplot.py:161)    
`

- If schematic or 3d model exports are not working, Environment variables may not be set up properly
`
ERROR:Potential dialog messages: {'F.Cu (PgUp)', 'B.Paste', 'Selection Filter', "Drawing sheet '${VARIABLE}/kicadpagelayout.kicad_wks' not found.", 'F.Adhesive', 'B.Adhesive', 'F.Paste', 'Appearance', 'B.Cu (PgDn)', 'OK'} (pcbnew_do - interposer.py:363) (kibot - kiplot.py:161)                                                                                                                 ERROR:pcbnew_do returned 9 (kibot - out_netlist.py:63)
`

**Comment by set-soft** [comment]
Hi @dbriscoe511 !
Thank you very much for sharing the steps needed to install KiBot using WSL.
Here are some details:
1) I suggest installing KiBot and KiAuto like this:

---
- Add the repo as a valid source for packages:

```shell
$ wget https://set-soft.github.io/debian/kibot.list
$ sudo cp kibot.list /etc/apt/sources.list.d/
```

- Add the repo signature to your system:

```shell
$ wget https://set-soft.github.io/debian/kibot.gpg
$ sudo cp kibot.gpg /etc/apt/
```

- Update your list of packages:

```shell
$ sudo apt-get update
```

- Install the main package:

```shell
$ sudo apt-get install kibot
```
---
This will install much more tools, and upgrading any tool will be as easy as upgrading any other Ubuntu tool.

2) How do you copy things between both *worlds*? I never used WSL and I don't have a Windows at hand.
You mention that you can ask Windows to open a Linux terminal at a desired place, what the filesystem looks like? I mean: can you access `C:\Users\` from Linux? How? Is this `/c/users/`?
Also: Is the username mapped to Linux as an environment variable?
I ask all of this because KiBot could simply check if you are using WSL and try loading the KiCad config from Windows if you don't have it.
Also: Can you create symlinks from the Linux space to the Windows space? What happens if you link the KiCad config dir?

3) The problem you are experimenting with the netlist seems to be related to the fact that KiCad can't find the worksheet for your project: `*.kicad_wks`

**Reply by set-soft** [reply]
I found some answers here: [Working across Windows and Linux file systems](https://docs.microsoft.com/en-us/windows/wsl/filesystems)

- How the Linux side sees Windows? The drives are just mounted in `/mnt/`, so `C:\` is `/mnt/c/`
- I don't understand why this document adds some magic `$`, do you know?
- According to the document running `wsl kibot` from a command prompt (or Powershell) should be enough to run KiBot at the current directory 
- The document says that you can even share environment variables using the `WSLENV` environment variable.

**Reply by dbriscoe511** [reply]
Thanks for the tips! I edited the post with the better install methods you suggested. 

I was getting two errors with the netlist output for a second, before I figured out how to set the environment variables. I accidentally pasted both into this discussion, but I updated my original post to clarify. The netlist seems to not work occasionally for me now, maybe 10% of the time, but restarting the shell seems to fix it reliably. 

The ${VARIABLE} notation is what kicad uses for it's environment variables, and will substitute the path in the environment variable if it can find it, or throw an error if it cannot. Sharing windows environment variables will not work, since these are kicad specific. If you add something in preferences->configure paths, this is an environment variable.

**Reply by set-soft** [reply]
> The netlist seems to not work occasionally for me now, maybe 10% of the time, but restarting the shell seems to fix it reliably.

This should be fixed by INTI-CMNB/KiAuto@212e2cbdc62f947e85c83e26f5ded3ed9f037675
The problem looks quite random because the "Info" dialog is created in an asynchronous way, KiCad starts, loads the schematic/PCB, and starts waiting for user input. In another thread KiCad starts to draw the schematic/PCB, but during the process it detects the lack of the .kicad_wks and displays the dialog. The problem is that we already started to interact with KiCad. So this is a race condition. Now KiAuto can detect those dialogs, press "Return" to close them and retry the operation. Seems to be working, but could need adjusts.

**Reply by set-soft** [reply]
> The ${VARIABLE} notation is what kicad uses for it's environment variables,

KiCad should also replace environment variables there. This isn't a KiCad notation, this is UNIX shell notation. KiCad can replace ${ENVIRONMENT_VARIABLE_NAME} in anything referring to a file. And you can define the environment variable from outside KiCad, in your shell. According to Microsoft environment variables can be shared between both.

**Comment by Unichord** [comment]
Hello,
I found this guide extremely useful, I managed to get KiBot to (almost) work.
For future reference, WSL uses all available RAM, so I limited it by creating a `.wslconfig` file in my user folder, I think 1 GB is enough to run only KiBot:
```
[wsl2]
memory=1GB
```

`kibot-check` works fine, as far as I can tell.

By doing a `kibot --quick-start` inside a project folder I get most of the outputs (BOMs, gerbers, PDFs, etc), but I'm having trouble with interoperability between Windows and WSL. This for example:
`/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/C:\Users\Andrea\kicad6\3d/Passive_SMD_3d/R_0603_1608Metric.stp`
Which was not solved by adding my 3D folder path to user environment variables, then `WSLENV=MY3DPATH/p`. In WSL I can see the environment variable set correctly, but it still doesn't work. Should I define this variable somewhere?

Other errors I'm getting are (also by running WSL as admin, and the command as sudo):
```
ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_navigate_results.py:141)
DEBUG:Output from command: convert: attempt to perform an operation not allowed by the security policy `PS' @ error/constitute.c/IsCoderAuthorized/408.
```
```
ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_navigate_results.py:141)
DEBUG:Output from command: convert: attempt to perform an operation not allowed by the security policy `PDF' @ error/constitute.c/IsCoderAuthorized/408.
```
```
ERROR:Failed to run /usr/bin/rsvg-convert, error 1 (kibot - out_navigate_results.py:141)
DEBUG:Output from command: Error reading SVG:XML parse error: Error domain 1 code 74 on line 5477 column 19 of data: EndTag: '</' not found
```
```
ERROR:/usr/bin/pandoc error: 47 (kibot - out_report.py:798)
DEBUG:Output from command: pdflatex not found. Please select a different --pdf-engine or install pdflatex -- see also /usr/share/doc/pandoc/README.Debian (kibot - out_report.py:800)
```
I guess the last one can be solved by installing pdflatex, but I was wondering if maybe a dependency was broken?

**Reply by set-soft** [reply]
Hi @Unichord !

Thanks for sharing your experience!

> I think 1 GB is enough to run only KiBot:

Be careful, KiBot doesn't need much memory, but KiCad must run there, and it can take a lot of memory (too much for my taste).

About the strange path: `/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/C:\Users\Andrea\kicad6\3d/Passive_SMD_3d/R_0603_1608Metric.stp`
We should take a look at the kibot output using `-vvv` to figure out how is constructed. I suspect the environment variables that configure KiCad are wrong.
My guess: The KiCad configuration makes reference to `C:\Users\Andrea\kicad6\3d` which doesn't look like an absolute path to KiBot. If this is some setting in the project you should look for some solution. May be using a path relative to the project, or an environment variable you can define in a way for Windows and in another way for Linux. This problem is more related to interoperability ... if you move your project to a Linux machine, or even another Windows machine where the user `Andrea` doesn't exist you'll have the same problem,

> convert: attempt to perform an operation not allowed by the security policy

This is documented in the "Installation - Dependencies" section:

Note that on some systems (i.e. Debian) ImageMagick disables PDF manipulation in its `policy.xml` file. Comment or remove lines like this: `<policy domain="coder" rights="none" pattern="PDF" />` (On Debian: `/etc/ImageMagick-6/policy.xml`)

**Reply by dbriscoe511** [reply]
This is related to kicad specific environment variables. The naming here is confusing, because the kicad ${variable} notation is the same as the unix shell notation, but the environment variables have nothing to do with the OS environment variables. This why importing environment variables from windows does not work.

You can see an example of this if you open a part in the PCB editor, and go to the 3d models tab.

To fix your error, you can open kicad in wsl if you have graphics set up and go to preferences - configure paths and change the C:\\\\ to mnt/c/  
This is the easiest way, editing the json below can be kind of convoluted. 

If you are unable to get graphics set up, you can edit the kicad settings json to set up the paths in ~/.config/kicad/6.0/kicad_common.json

The setup is the same for customized paths to libraries.

**Reply by Unichord** [reply]
Hello, thanks both for your help.

So, my libraries are not defined in an absolute way, but like this: `${KICAD6_FOOTPRINT_CLE}/Passive_SMD.pretty`. In "Configure paths" obviously they're defined in an absolute way. Since I don't have access to graphics on WSL, I imported the kicad_common.json from Windows as said in the original post, and I edited it to make all paths look like `/mnt/c/Users/Andrea/kicad6/3d`. It now appears to work.

About the memory needed: I kept monitoring the used memory by `free -h` and I noticed that, in my case, it never goes over 600 MB when running KiBot, and only on most intensive tasks (e.g. renders). So, to be sure, maybe reserve 2 GB, but I don't think it will be necessary.

About the `convert` error: yes I missed the policy.xml part, sorry. I commented out all lines like that as you suggested (PS, PDF, EPS), it seemed to have improved but I still get errors about policy.

I also have some other errors, I don't know if they're related:
```
ERROR:/usr/bin/pandoc error: 47 (kibot - out_report.py:798)
DEBUG:Output from command: pdflatex not found. Please select a different --pdf-engine or install pdflatex -- see also /usr/share/doc/pandoc/README.Debian
 (kibot - out_report.py:800)
```
```
ERROR:/usr/bin/pandoc error: 47 (kibot - out_report.py:798)
DEBUG:Output from command: [WARNING] Could not fetch resource 'Schematic/FMFR1-schematic.svg': replacing image with description
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.070: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.095: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.103: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.105: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.105: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.105: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28313): Pango-CRITICAL **: 09:51:42.126: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28314): Pango-CRITICAL **: 09:51:42.494: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28314): Pango-CRITICAL **: 09:51:42.501: pango_font_description_set_size: assertion 'size >= 0' failed
(rsvg-convert:28314): Pango-CRITICAL **: 09:51:42.513: pango_font_description_set_size: assertion 'size >= 0' failed
pdflatex not found. Please select a different --pdf-engine or install pdflatex -- see also /usr/share/doc/pandoc/README.Debian
 (kibot - out_report.py:800)
```
I installed `texlive-latex-base` and `texlive-latex-recommended` and tried running the command `pdflatex` in the WSL shell and it does something, so I don't know what could be the problem.

```
ERROR:Failed to create Step file, error 255 (kibot - out_step.py:100)
DEBUG:Output from command: DEBUG:Detected KiCad v6.0.9 (6.0.9-8da3e8f707~116~ubuntu20.04.1 6000009) (kicad2step_do - misc.py:146)
DEBUG:Config path /home/andrea/.config/kicad/6.0 (kicad2step_do - misc.py:162)
DEBUG:KiCad environment: {'KICAD6_3D_CLE': 'C:\\Users\\Andrea\\kicad6\\3d', 'KICAD6_FOOTPRINT_CLE': 'C:\\Users\\Andrea\\kicad6\\footprint', 'KICAD6_SYMBOL_CLE': 'C:\\Users\\Andrea\\kicad6\\symbol', 'KICAD6_TEMPLATE_CLE': 'C:\\Users\\Andrea\\kicad6\\template', 'KICAD_USER_TEMPLATE_DIR': 'C:\\Users\\Andrea\\Documents\\kicad\\template', 'KIGITHUB': 'https://github.com/KiCad', 'SAMACSYS_3D': 'C:\\Users\\Andrea\\kicad\\samacsys\\3d', 'SAMACSYS_LIB': 'C:\\Users\\Andrea\\kicad\\samacsys'} (kicad2step_do - misc.py:252)
DEBUG:Output dir: /mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/3D (kicad2step_do - kicad2step_do:117)
```
Which is weird, since I edited `kicad_common.json`. Where else can I look for these variables? In the `.config` folder I already checked, there are no other places where these variables are defined.

**Reply by set-soft** [reply]
Hi @Unichord !

> I also have some other errors, I don't know if they're related:

Installing `pandoc` is tricky. Perhaps some missing fonts.
Note that `rsvg-convert` is also saying something about the fonts. 
What do you get when running `convert -list font`? Do you have some Helvetica replacement?

> Where else can I look for these variables?

Try `export | grep KICAD`. The values should be read from `/home/andrea/.config/kicad/6.0/kicad_common.json`, from the `environment.vars` section.

**Reply by Unichord** [reply]
Hi @set-soft, thanks again for the support, sorry but I'm not super familiar with Linux.

This is what I get about Helvetica when running `convert -list font`, I can upload the whole file if needed:
```
  Font: fixed
    family: Helvetica
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: /usr/share/fonts/type1/gsfonts/n019003l.pfb
  Font: Helvetica
    family: Helvetica
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: /usr/share/fonts/type1/gsfonts/n019003l.pfb
  Font: Helvetica-Bold
    family: Helvetica
    style: Normal
    stretch: Normal
    weight: 700
    glyphs: /usr/share/fonts/type1/gsfonts/n019004l.pfb
  Font: Helvetica-BoldOblique
    family: Helvetica
    style: Italic
    stretch: Normal
    weight: 700
    glyphs: /usr/share/fonts/type1/gsfonts/n019024l.pfb
  Font: Helvetica-Narrow
    family: Helvetica Narrow
    style: Normal
    stretch: Condensed
    weight: 400
    glyphs: /usr/share/fonts/type1/gsfonts/n019043l.pfb
  Font: Helvetica-Narrow-Bold
    family: Helvetica Narrow
    style: Normal
    stretch: Condensed
    weight: 700
    glyphs: /usr/share/fonts/type1/gsfonts/n019044l.pfb
  Font: Helvetica-Narrow-BoldOblique
    family: Helvetica Narrow
    style: Oblique
    stretch: Condensed
    weight: 700
    glyphs: /usr/share/fonts/type1/gsfonts/n019064l.pfb
  Font: Helvetica-Narrow-Oblique
    family: Helvetica Narrow
    style: Oblique
    stretch: Condensed
    weight: 400
    glyphs: /usr/share/fonts/type1/gsfonts/n019063l.pfb
  Font: Helvetica-Oblique
    family: Helvetica
    style: Italic
    stretch: Normal
    weight: 400
    glyphs: /usr/share/fonts/type1/gsfonts/n019023l.pfb
```
These fonts are not missing in that location.

When running `export | grep KICAD` I don't get all the variables defined in `kicad_common.json`, but only the ones I defined manually when trying to import Windows environment variables:
```
declare -x KICAD6_3D_CLE="/mnt/c/Users/Andrea/kicad6/3d"
declare -x WSLENV="KICAD6_3D_CLE/p"
```

**Reply by Unichord** [reply]
I'm making some progress.

Turns out `texlive` was missing, so I ran:
```
sudo apt-get install texlive
sudo apt-get install texlive-latex-extra
```
This solved pdf generation problems.

For future reference, this is what my `\etc\ImageMagick-6\policy.xml` looks like:
```
<!-- use curl -->
  <policy domain="delegate" rights="none" pattern="URL" />
  <policy domain="delegate" rights="none" pattern="HTTPS" />
  <policy domain="delegate" rights="none" pattern="HTTP" />
  <!-- in order to avoid to get image with password text -->
  <policy domain="path" rights="none" pattern="@*" />
  <!-- disable ghostscript format types -->
  <!--policy domain="coder" rights="none" pattern="PS" /-->
  <policy domain="coder" rights="none" pattern="PS2" />
  <policy domain="coder" rights="none" pattern="PS3" />
  <!--policy domain="coder" rights="none" pattern="EPS" /-->
  <!--policy domain="coder" rights="none" pattern="PDF" /-->
  <!--policy domain="coder" rights="none" pattern="XPS" />
```

KiBot is already usable as is, but I'm still getting this error:
```
ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_pcb_print.py:83)
DEBUG:Output from command: convert: no images defined `/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/PCB/PNG/FMFR1-assembly_page_01.png' @ error/convert.c/ConvertImageCommand/3258.
 (kibot - out_pcb_print.py:85)
```
Which leads to a half-generated image (this is a small sample):
![image](https://user-images.githubusercontent.com/73905953/201320280-abc2ecc8-1829-4a3a-af50-e656c29be35a.png)

And also this error during stp model generation, sorry for the wall of text, I don't know how to limit the number of lines:
```
ERROR:Failed to create Step file, error 255 (kibot - out_step.py:100)
DEBUG:Output from command: DEBUG:Detected KiCad v6.0.9 (6.0.9-8da3e8f707~116~ubuntu20.04.1 6000009) (kicad2step_do - misc.py:146)
DEBUG:Config path /home/andrea/.config/kicad/6.0 (kicad2step_do - misc.py:162)
DEBUG:KiCad environment: {'KICAD6_3D_CLE': '/mnt/c/Users/Andrea/kicad6/3d', 'KICAD6_FOOTPRINT_CLE': '/mnt/c/Users/Andrea/kicad6/footprint', 'KICAD6_SYMBOL_CLE': '/mnt/c/Users/Andrea/kicad6/symbol', 'KICAD6_TEMPLATE_CLE': '/mnt/c/Users/Andrea/kicad6/template', 'KICAD_USER_TEMPLATE_DIR': '/mnt/c/Users/Andrea/Documents/kicad/template', 'KIGITHUB': 'https://github.com/KiCad', 'SAMACSYS_3D': '/mnt/c/Users/Andrea/kicad/samacsys/3d', 'SAMACSYS_LIB': '/mnt/c/Users/Andrea/kicad/samacsys'} (kicad2step_do - misc.py:252)
DEBUG:Output dir: /mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/3D (kicad2step_do - kicad2step_do:117)
INFO:KiCad to STEP wrapper
DEBUG:Command: ['kicad2step', '-o', '/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/3D/FMFR1-3D.step', '-f', '--grid-origin', '--subst-models', '/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/tmpfrq2dad7.kicad_pcb'] (kicad2step_do - kicad2step_do:144)
DEBUG:Redirecting kicad2step output to kicad2step*.log (kicad2step_do.kiauto.file_util - file_util.py:361)
DEBUG:Waiting for virtual X server ... (kicad2step_do.kiauto.ui_automation - ui_automation.py:91)
DEBUG:Current DISPLAY is :1551580448 (kicad2step_do.kiauto.ui_automation - ui_automation.py:93)
DEBUG:Checking using ['setxkbmap', '-query'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:104)
DEBUG:Redirecting setxkbmap output to setxkbmap*.log (kicad2step_do.kiauto.file_util - file_util.py:361)
DEBUG:['setxkbmap', '-query'] returned 0 (kicad2step_do.kiauto.ui_automation - ui_automation.py:110)
DEBUG:Waiting 2.0 seconds before using the X server (kicad2step_do.kiauto.ui_automation - ui_automation.py:111)
DEBUG:Recording session with: ['recordmydesktop', '--overwrite', '--no-sound', '--no-frame', '--on-the-fly-encoding', '-o', '/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/3D/kicad2step_screencast.ogv'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:166)
DEBUG:Redirecting recordmydesktop output to recordmydesktop*.log (kicad2step_do.kiauto.file_util - file_util.py:361)
INFO:Waiting for "KiCad to STEP" ...
DEBUG:['xdotool', 'search', '--onlyvisible', '--name', '.*'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'getwindowname', '640'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:get_windows 640  (kicad2step_do.kiauto.ui_automation - ui_automation.py:283)
DEBUG:get_windows end of list (kicad2step_do.kiauto.ui_automation - ui_automation.py:285)
DEBUG:['xdotool', 'search', '--onlyvisible', '--name', '.*'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'getwindowname', '640'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:get_windows 640  (kicad2step_do.kiauto.ui_automation - ui_automation.py:283)
DEBUG:get_windows end of list (kicad2step_do.kiauto.ui_automation - ui_automation.py:285)
DEBUG:['xdotool', 'search', '--onlyvisible', '--name', '.*'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'getwindowname', '640'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:get_windows 640  (kicad2step_do.kiauto.ui_automation - ui_automation.py:283)
DEBUG:['xdotool', 'getwindowname', '4194311'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:get_windows 4194311 Kicad2step (kicad2step_do.kiauto.ui_automation - ui_automation.py:283)
DEBUG:['xdotool', 'getwindowname', '4194312'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:get_windows 4194312  (kicad2step_do.kiauto.ui_automation - ui_automation.py:283)
DEBUG:get_windows end of list (kicad2step_do.kiauto.ui_automation - ui_automation.py:285)
DEBUG:Found KiCad to STEP window (1) (kicad2step_do.kiauto.ui_automation - ui_automation.py:381)
DEBUG:Window id: 4194311 (kicad2step_do.kiauto.ui_automation - ui_automation.py:386)
DEBUG:['xdotool', 'windowfocus', '--sync', '4194311'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Waiting for 4194311 window to get focus... (kicad2step_do.kiauto.ui_automation - ui_automation.py:317)
DEBUG:['xdotool', 'getwindowfocus'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Currently focused id: 4194311 (kicad2step_do.kiauto.ui_automation - ui_automation.py:320)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Error: target STRING not available
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
DEBUG:['xdotool', 'click', '1'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:['xdotool', 'key', 'ctrl+a', 'ctrl+c'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Clipboard retrieve "Read file: '/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/tmpfrq2dad7.kicad_pcb'
Build STEP data
Add component C1.
[...]
Add component RL1.
 Errors on Reading Header Entity N0.5:HeaderSection_FileName
Parameter n0.7 (authorisation) not a quoted String

Add component RL2.
[...]
Add component R2.
Create PCB solid model
Build board outline (11 items).
Could not close outline (dropping outline data with 1 segments).
 + line start: 153.67,-133.858 end: 85.852,-133.858
Could not create closed board outlines.
could not create PCB solid model

** Error building STEP board model. Export aborted. **
" (kicad2step_do.kiauto.ui_automation - ui_automation.py:265)
INFO:Read file: '/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/tmpfrq2dad7.kicad_pcb'
Build STEP data
Add component C1.
[...]
Add component RL1.
 Errors on Reading Header Entity N0.5:HeaderSection_FileName
Parameter n0.7 (authorisation) not a quoted String

Add component RL2.
[...]
Add component R2.
Create PCB solid model
Build board outline (11 items).
Could not close outline (dropping outline data with 1 segments).
 + line start: 153.67,-133.858 end: 85.852,-133.858
Could not create closed board outlines.
could not create PCB solid model

** Error building STEP board model. Export aborted. **

DEBUG:['xdotool', 'key', 'Tab', 'Return'] (kicad2step_do.kiauto.ui_automation - ui_automation.py:218)
DEBUG:Closing pipe with 15071 (kicad2step_do.kiauto.ui_automation - ui_automation.py:48)
DEBUG:Terminating the session recorder (kicad2step_do.kiauto.ui_automation - ui_automation.py:172)
DEBUG:Closing pipe with 15070 (kicad2step_do.kiauto.ui_automation - ui_automation.py:48)
```

**Reply by Unichord** [reply]
> ```
> ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_pcb_print.py:83)
> DEBUG:Output from command: convert: no images defined `/mnt/c/Users/Andrea/Documents/Progetti/CLE/FMFR1_PROVA_6.0/Generated/PCB/PNG/FMFR1-assembly_page_01.png' @ error/convert.c/ConvertImageCommand/3258.
>  (kibot - out_pcb_print.py:85)
> ```
> 
> Which leads to a half-generated image (this is a small sample): ![image](https://user-images.githubusercontent.com/73905953/201320280-abc2ecc8-1829-4a3a-af50-e656c29be35a.png)

I did some more tinkering, and I saw that the half-generated image is not the consequence of the error, but the cause. So I tried to delete the entire `Generated` folder, and magic! No more errors.

While running again `kibot --quick-start` without deleting the folder, I don't get the "Failed to create Step file" error anymore, but I still get the "convert" error. I believe it's due to the half-generated PNG images, because if I delete only the `/Generated/PCB/PNG/` folder and run `kibot --quick-start` I don't get any error and the PNG images are generated fully and correctly.

@set-soft I consider my problems solved, if you need more information to debug and possibly fix stuff (I don't really know where the problem is), I can do some more tests, let me know.

**Reply by set-soft** [reply]
Hi @Unichord !
Thanks for the info.
I added `texlive` as suggested package for `pandoc`. Looking at its dependencies it looks like this is needed for the fonts.
The `convert` error with partial image is strange. It looks like KiBot is converting the PDF to PNG before the PDF is finished. But we wait until ghostscript finishes ... so this is bizarre.
